src/LiveQuery/QueryTools.js

```diff
@@ -133,6 +133,13 @@ function matchesKeyConstraints(object, key, constraints) {
   if (constraints === null) {
     return false;
   }
+  if(key.indexOf(".") >= 0){
```

Comment:
No, it isn't, but I didn't see anywhere that invalid keys should be screened out. In this case, it will recursively call `matchesKeyConstraints` with an empty object, which should result in the same result as what this code was doing before with an invalid key. You could make the comparison `>` instead of `>=`, if you think that makes it more clear, but it should ultimately return the same answer.